dbms/src/AggregateFunctions/AggregateFunctionCategoricalInformationValue.h

```diff
@@ -0,0 +1,156 @@
+    {
+        // notice: argument types has been checked before
+    }
+
+    String getName() const override
+    {
+        return "categoricalInformationValue";
+    }
+
+    const char * getHeaderFilePath() const override
+    {
+        return __FILE__;
+    }
+
+    void create(AggregateDataPtr place) const override
+    {
+        memset(place, 0, sizeOfData());
+    }
+
+    void destroy(AggregateDataPtr) const noexcept override
+    {
+        // nothing
+    }
+
+    bool hasTrivialDestructor() const override
+    {
+        return true;
+    }
+
+    size_t sizeOfData() const override
+    {
+        return sizeof(T) * (category_count + 1) * 2;
+    }
+
+    size_t alignOfData() const override
+    {
+        return alignof(T);
+    }
+
+    void add(
+        AggregateDataPtr place,
+        const IColumn ** columns,
+        size_t row_num,
+        Arena *
+    ) const override
+    {
+        auto y_col = static_cast<const ColumnUInt8 *>(columns[category_count]);
+        bool y = y_col->getData()[row_num];
+
+        for (size_t i : ext::range(0, category_count))
+        {
+            auto x_col = static_cast<const ColumnUInt8 *>(columns[i]);
+            bool x = x_col->getData()[row_num];
+
+            if (x)
+                reinterpret_cast<T *>(place)[i * 2 + size_t(y)] += 1;
+        }
+
+        reinterpret_cast<T *>(place)[category_count * 2 + size_t(y)] += 1;
+    }
+
+    void merge(
+        AggregateDataPtr place,
+        ConstAggregateDataPtr rhs,
+        Arena *
+    ) const override
+    {
+        for (size_t i : ext::range(0, category_count + 1))
+        {
+            reinterpret_cast<T *>(place)[i * 2] += reinterpret_cast<const T *>(rhs)[i * 2];
+            reinterpret_cast<T *>(place)[i * 2 + 1] += reinterpret_cast<const T *>(rhs)[i * 2 + 1];
+        }
+    }
+
+    void serialize(
+        ConstAggregateDataPtr place,
+        WriteBuffer & buf
+    ) const override
+    {
+        buf.write(place, sizeOfData());
+    }
+
+    void deserialize(
+        AggregateDataPtr place,
+        ReadBuffer & buf,
+        Arena *
+    ) const override
+    {
+        buf.read(place, sizeOfData());
+    }
+
+    DataTypePtr getReturnType() const override
+    {
+        return std::make_shared<DataTypeArray>(
+            std::make_shared<DataTypeNumber<Float64>>()
+        );
+    }
+
+    void insertResultInto(
+        ConstAggregateDataPtr place,
+        IColumn & to
+    ) const override
+    {
+        auto & col = static_cast<ColumnArray &>(to);
+        auto & data_col = static_cast<ColumnFloat64 &>(col.getData());
+        auto & offset_col = static_cast<ColumnArray::ColumnOffsets &>(
+            col.getOffsetsColumn()
+        );
+
+        data_col.reserve(data_col.size() + category_count);
+
+        T sum_no = reinterpret_cast<const T *>(place)[category_count * 2];
+        T sum_yes = reinterpret_cast<const T *>(place)[category_count * 2 + 1];
+
+        Float64 rev_no = 1. / sum_no;
```

Comment:
Can we understand is it enough to use just float here? I don't actually know with what precision do we want such evaluations, but if we can allow rounding it would be interesting to have some experiments....

Reply:
It's Ok.

For example, we always use Float64 for `avg`.
And this function is by idea similar to `entropy` that is also using Float64.